gui/wxpython/gis_set.py

```diff
@@ -510,9 +511,11 @@ def SuggestDatabase(self):
         if self.GetRCValue("LOCATION_NAME") != "<UNKNOWN>":
             return
         path = get_possible_database_path()
-        # If nothing found, try to create GRASS directory
+
+        # If nothing found, try to create GRASS directory and copy startup loc
         if path is None:
             path = create_database_directory()
+            location = get_startup_location(path)
```

Comment:
Since you added the mapset part in this PR, let's continue here. Now the mapset is not used and the user is presented with two mapsets to choose from. This is a situation we want to avoid. Let's take then the final steps from `OnStart()` and set the mapset and exit the startup with:

```
        self.SetLocation(dbase, location, mapset)
        self.ExitSuccessfully()
```

This will achieve the final/actual goal of #862 (see the title).

Reply:
These two lines are now there.